action/protocol/staking/typedcandidate.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package staking
+
+import (
+	"math/big"
+
+	"github.com/iotexproject/iotex-address/address"
+)
+
+type (
+	// typedCandidateRegister is the interface that every candidate register action should implement
```

Comment:
private or public?

Reply:
it's a interface used within this package, so should be private